File: leetcodeJS/lists/206.reverse-linked-list.js
function ListNode(val, next) {
  this.val = val === undefined ? 0 : val;
  this.next = next === undefined ? null : next;
}
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function (head) {
  if (!head) return null;
  let arr = [];
  while (head) {
    arr.push(head.val);
    head = head.next;
  }

  head = new ListNode(arr.pop());
  let temp = head;
  for (let i = arr.length - 1; i >= 0; i--) {
    head.next = new ListNode(arr[i]);
    head = head.next;
  }
  return temp;
};

File: leetcodePython/lists/206.reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        if not head:
            return
        arr = []
        while head:
            arr.append(head.val)
            head = head.next

        head = ListNode(arr.pop())
        node = head
        for i in range(len(arr)-1, -1, -1):
            head.next = ListNode(arr[i])
            head = head.next

        return node
